src/lib/ts/tsconfig.ts

```diff
@@ -50,55 +71,43 @@ export const initializeTsConfig = (defaultTsConfig: TsConfig, entryPoints: Entry
 
-    // minimal compilerOptions needed in order to avoid errors, with their associated default values
-    // some are not overrided in order to keep the default associated TS errors if the user choose to set incorrect values
-    const requiredOptions: Partial<ts.CompilerOptions> = {
-      emitDecoratorMetadata: true,
-      experimentalDecorators: true,
-      moduleResolution: ts.ModuleResolutionKind.NodeJs,
-      target: ts.ScriptTarget.ES2015,
-      declaration: true,
-      lib: ['dom', 'es2015']
-    };
-
-    const overrideConfig: Partial<TsConfig> = {
-      rootNames: [entryPoint.entryFilePath],
-      options: {
-        flatModuleId: entryPoint.moduleId,
-        flatModuleOutFile: `${entryPoint.flatModuleFile}.js`,
-        basePath: basePath,
-        rootDir: basePath,
-        outDir: '',
-        lib: entryPoint.languageLevel ? entryPoint.languageLevel.map(lib => `lib.${lib}.d.ts`) : tsConfig.options.lib,
-        // setting this as basedir will rewire triple-slash references
-        declarationDir: basePath,
-        // required in order to avoid "ENOENT: no such file or directory, .../.ng_pkg_build/..." errors when using the programmatic API
-        inlineSources: true,
-        // setting the below here because these are a must have with these valus
-        inlineSourceMap: true,
-        sourceMap: false,
-        sourceRoot: `ng://${entryPoint.moduleId}`,
-        enableResourceInlining: true
-      }
-    };
-
-    tsConfig.rootNames = overrideConfig.rootNames;
-    tsConfig.options = { ...requiredOptions, ...tsConfig.options, ...overrideConfig.options };
-
+    let jsx = tsConfig.options.jsx;
     switch (entryPoint.jsxConfig) {
       case 'preserve':
-        tsConfig.options.jsx = ts.JsxEmit.Preserve;
+        jsx = ts.JsxEmit.Preserve;
         break;
       case 'react':
-        tsConfig.options.jsx = ts.JsxEmit.React;
+        jsx = ts.JsxEmit.React;
         break;
       case 'react-native':
-        tsConfig.options.jsx = ts.JsxEmit.ReactNative;
+        jsx = ts.JsxEmit.ReactNative;
         break;
       default:
         break;
     }
 
+    const overrideOptions: ng.CompilerOptions = {
+      flatModuleId: entryPoint.moduleId,
+      flatModuleOutFile: `${entryPoint.flatModuleFile}.js`,
+      basePath,
+      rootDir: basePath,
+      lib: entryPoint.languageLevel ? entryPoint.languageLevel.map(lib => `lib.${lib}.d.ts`) : tsConfig.options.lib,
+      declarationDir: basePath,
+      sourceRoot: `ng://${entryPoint.moduleId}`,
+      jsx
+    };
+
+    tsConfig.rootNames = [entryPoint.entryFilePath];
+    tsConfig.options = { ...tsConfig.options, ...overrideOptions };
+
+    // Add paths mappings for dependencies
+    const entryPointDeps = entryPoints.filter(x => x.data.entryPoint.moduleId !== entryPoint.moduleId);
+    if (entryPointDeps.length > 0) {
+      if (!tsConfig.options.paths) {
+        tsConfig.options.paths = {};
+      }
+    }
+
     currentEntryPoint.data.tsConfig = tsConfig;
```

Comment:
Had merge conflicts here. Are the conflicts resolved properly?

Reply:
No, I will fix them now

Reply:
@dherges conflicts should be good now, waiting for CI to go green.

Thanks anyways for trying 👍